modules/sdk-coin-apt/src/lib/transactionBuilder/transactionBuilder.ts

```diff
@@ -37,6 +45,8 @@ export abstract class TransactionBuilder extends BaseTransactionBuilder {
     this._transaction = transaction;
   }
 
+  abstract validateTransaction(transaction?: Transaction): void;
```

Comment:
doesn't need to be an abstract function.
Add the common part of the implementation in the parent class and override if there is additional validation needed in a child class.